python/ray/dashboard/modules/dashboard_sdk.py

```diff
@@ -242,6 +249,22 @@ def __init__(
             else:
                 self._ssl_context = None
 
+    def _set_auth_header_if_enabled(self):
+        """Add authentication token to headers if token auth is enabled."""
+        if is_token_auth_enabled():
+            token_loader = AuthenticationTokenLoader.instance()
+            token_added = token_loader.set_token_for_http_header(self._headers)
```

Comment:
same thing here -- better to have the `AuthenticationTokenLoader` return the headers instead of modify an argument passed by value